### PR TITLE
Feature: Custom loss function and Tensorboard

### PR DESCRIPTION
**Pull request by hlfshell**
This PR introduces two major changes to the training.

The first is the introduction of an additional loss function. This loss function is Sparse Categorical Cross Entropy Focal Loss. We pair it with weights calculated based on averages and percentages of given category pixel counts throughout the dataset.

It also introduces tensorboard for graph creation and progress tracking during and after the training process.

### CODE REVIEW

**Review by Rhodwarrior — APPROVED** [review]
DIdnt see necessary changes

**Review by zekeflaton — APPROVED** [review]
